plutus-pab/src/Cardano/Wallet/Mock.hs

```diff
@@ -91,11 +93,17 @@ handleMultiWallet = do
             Seed seed <- generateSeed
             let bytes = BS.pack . unpack $ seed
             let privateKey = PrivateKey (KB.fromBytes bytes)
-            let pkh = pubKeyHash (toPublicKey privateKey)
+            let pubKey = toPublicKey privateKey
+            let pkh = pubKeyHash pubKey
             let walletId = byteString2Integer (getPubKeyHash pkh)
             let wallet = Wallet walletId
             let wallets' = Map.insert wallet privateKey wallets
             put wallets'
+            let walletState = WalletState privateKey2 emptyNodeClientState mempty (defaultSigningProcess (Wallet 2))
```

Comment:
why `privateKey2`? And have you tested this? I'm not sure wallet 2 will know which funds to use when it runs the action later on.

Reply:
That's a weird one. It didn't work with Wallet 1/privKey1. But it works with privKey2, I tested it.
Don't know why, but I want to move to companion contract asap.

Reply:
I tried to create multiple wallets, and it worked semi-fine. It create 2 or 3 successfully, and then the REST API call just hanged.